docker/android-ndk.sh

```diff
@@ -6,26 +6,59 @@ set -euo pipefail
 # shellcheck disable=SC1091
 . lib.sh
 
-NDK_URL=https://dl.google.com/android/repository/android-ndk-r21d-linux-x86_64.zip
-
 main() {
-    local arch="${1}" \
-        api="${2}"
+    local arch="${1}"
+
+    if [[ "${ANDROID_SDK}" -eq 25 ]]; then
+        echo "Android SDK level 25 is not supported." 1>&2
+        exit 1
+    fi
 
-    install_packages curl unzip python
+    # python3 is still needed for newer NDK versions, just since it
+    # simplifies making symlinks even though the toolchain is prebuilt
+    install_packages curl python python3
+    get_ndk_info
+    if [[ "${NDK_VERSION}" -le 9 ]]; then
+        install_packages bzip2
+    else
+        install_packages unzip
+    fi
 
     local td
     td="$(mktemp -d)"
 
     pushd "${td}"
     curl --retry 3 -sSfL "${NDK_URL}" -O
-    unzip -q android-ndk-*.zip
-    rm android-ndk-*.zip
-    pushd android-ndk-*
-    ./build/tools/make_standalone_toolchain.py \
-        --install-dir /android-ndk \
-        --arch "${arch}" \
-        --api "${api}"
+    if [[ "${NDK_VERSION}" -le 9 ]]; then
+        tar -xjf "${NDK_FILENAME}"
+    else
+        unzip -q "${NDK_FILENAME}"
+    fi
+    rm "${NDK_FILENAME}"
+    pushd "android-ndk-${ANDROID_NDK}"
+    # android NDK versions <= 13 error without the verbose flag
+    local build_cmd=
+    local api=
+    if [[ "${NDK_VERSION}" -le 11 ]]; then
```

Comment:
Newer NDK versions migrated from a shell script to a Python module. They don't do any builds, however, so the dependencies are the same.